shared/js/background/utils.es6.js

```diff
@@ -201,20 +201,35 @@ function isSafeListed (url) {
     return false
 }
 
+/**
+ * Tests whether the two URL's belong to the same
+ * first party set.
+ */
+function isFirstParty (url1, url2) {
+    const first = tldts.parse(url1)
+    const second = tldts.parse(url2)
+
+    const firstDomain = first.isIp ? first.hostname : first.domain
```

Comment:
we are not doing the `|| hostname` thing here 👀 should we?

Reply:
Looks like it - I added tests and there is an edge case that fails due to this: https://github.com/duckduckgo/duckduckgo-privacy-extension/runs/2164976053?check_suite_focus=true#step:5:296

Should we also using private PSL entries here too to align origin boundaries with the browser?

Reply:
Fixed by https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/570/commits/b1763ff0f218447493bfa68d62164b55a2abef51